tardis/model/base.py

```diff
@@ -133,7 +133,9 @@ def dilution_factor(self):
     @dilution_factor.setter
     def dilution_factor(self, new_dilution_factor):
         if len(new_dilution_factor) == self.no_of_shells:
-            self.radiation_field_state.dilution_factor = new_dilution_factor
+            self.radiation_field_state.dilution_factor[
+                self.geometry.v_inner_boundary_index : self.geometry.v_outer_boundary_index
+            ] = new_dilution_factor
```

Comment:
Probably should have a test for the case that the v_inner_boundary is in between shells